rpc/core/blocks.go

```diff
@@ -171,8 +172,12 @@ func BlockSearchMatchEvents(
 	if _, ok := env.BlockIndexer.(*blockidxnull.BlockerIndexer); ok {
 		return nil, errors.New("block indexing (match events) is disabled")
 	}
+
+	re := regexp.MustCompile(`^match.events[" "]?=[" "]?[0-1]$|match.events[" "]?=[" "]?[0-1] AND | AND match.events[" "]?=[" "]?[0-1]`)
```

Comment:
```suggestion
	re := regexp.MustCompile(`^match.events\s?=\s?[0-1]$|match.events\s?=\s?[0-1]\s+AND\s+|\s+AND match.events\s?=\s?[0-1]`)
```

Reply:
Also I think this justifies a couple extra UTs to test the regexp with weird values

Reply:
Instead of using regular expressions, which are pretty brittle and could introduce a substantial performance hit, why not just deduplicate the `match.events` conditions from the parsed query conditions before this call: https://github.com/tendermint/tendermint/blob/4c549b582d4f713fde7d5d55fb7cf0618e633e0a/state/txindex/kv/kv.go#L225

Reply:
Actually you read my mind, I changed this today but wanted to write a test or two to verify. Pushing this now.